src/test/java/PagesIvaF/BasePageIvan.java

```diff
@@ -0,0 +1,59 @@
+package PagesIvaF;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import javax.swing.*;
+import java.time.Duration;
+
+public class BasePageIvan {
+
+     WebDriver driver;
+     WebDriverWait wait;
+     Actions actions;
+    By allSongLocator = By.cssSelector("li a.songs");
+    By songVisaulizer = By.cssSelector("[data-testid = 'sound-bar-play']");
+     public BasePageIvan (WebDriver givenDriver){
+         driver=givenDriver;
+         wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+         actions = new Actions(driver);
+         PageFactory.initElements(driver, this);
+     }
+
+
+
+
+
+     public AllSongsIvan clickOnAllSongs ()  {
+//         WebElement loadingScreen = driver.findElement(By.cssSelector("#overlay"));
+//         wait.until(ExpectedConditions.invisibilityOf(loadingScreen));
+//then click all songs
+         wait.until(ExpectedConditions.visibilityOfElementLocated(allSongLocator));
+         driver.findElement(allSongLocator).click();
+         return new AllSongsIvan(driver);
+
+
+     }
+    public boolean isSongPlaying(){
+        wait.until(ExpectedConditions.visibilityOfElementLocated(songVisaulizer));
+        WebElement songVisualizer1 = driver.findElement(songVisaulizer);
+        return songVisualizer1.isDisplayed();
+    }
+    public void getContextMenu() {
+        By songLocator = By.cssSelector(".all-songs tr.song-item:nth-child(1)");
+        wait.until(ExpectedConditions.visibilityOfElementLocated(songLocator));
+        WebElement firstSong = driver.findElement(songLocator);
+        actions.contextClick(firstSong).perform();
+    }
+    public BasePageIvan choosePlayListByName(String name) {
+        By playListLocator = By.xpath( "//a[contains(text(), '" + name + "')]");
+        wait.until(ExpectedConditions.visibilityOfElementLocated(playListLocator)).click();
+        return this;
+    }
```

Comment:
**Base Page \ Base Class** -  Implements the necessary methods for work with webdriver.

We will store the web elements present on each page in the corresponding java class.

The methods `clickOnAllSongs() `  , `isSongPlaying()`  , `getContextMenu()`  , `choosePlayListByName`  must be stored in the class HomePage because the avatar is visible on the web site -home page

For example, all these methods should be in the class HomePage

![image](https://user-images.githubusercontent.com/104761482/202379599-dfcfc900-ea1f-4e67-bd0f-6fae9b69db38.png)